pkg/plugins/scms/github/main.go

```diff
@@ -70,7 +70,7 @@ func New(s Spec, pipelineID string) (*Github, error) {
 	}
 
 	if s.Directory == "" {
-		s.Directory = path.Join(tmp.Directory, s.Owner, s.Repository)
+		s.Directory = path.Join(tmp.Directory, "github", s.Owner, s.Repository)
```

Comment:
What is the reason of this change? It feels weird

Reply:
I made a comment in the PR description

_To avoid collision between gitea and github scm temporary directory, it now contains the service.
So github will clone repositories in /tmp/updatecli/github/... and gitea in /tmp/updatecli/gitea/..._

Reply:
Another option would be to do like git scm and creating the directory `github_com_updatecli_updatecli` and `try_gitea_io_updatecli_updatecli`

Reply:
In facts ,I should have add this comment to the code

Reply:
Why would there be a collision ? I’m not sure to see the difference with current updatecli manifest with different scm defined ?

Reply:
But a good refactoring would be to decide the path outside the scm plugin

Reply:
Interesting case: this is a bug and should be treated outside this PR as well honestly.

=> proposal: wdyt about using the SCM identifier as it would NEVER be possible to have the same values (instead of the type that could still lead to collisions?

eg. with your manifest:

```
/tmp/updatecli/example1/updatecli/updatecli
/tmp/updatecli/example2/updatecli/updatecli
```

Reply:
> Interesting case: this is a bug and should be treated outside this PR as well honestly.

IMHO I wouldn't consider it as a bug because it's a side effect introduced by this pullrequest.

While I am open to use a different schema, using the scmid seems wrong to me.
We commonly use the same scmid accross manifest such as default, local,etc
So the risk is higher.

Another option would be to use the same approach than the git repository so for example `https://github.com/updatecli/updatecli.git` becomes `/tmp/updatecli/https_github_com_updatecli_updatecli_git`

Reply:
> IMHO I wouldn't consider it as a bug because it's a side effect introduced by this pullrequest.

Fair point :)

> While I am open to use a different schema, using the scmid seems wrong to me. We commonly use the same scmid accross manifest such as default, local,etc So the risk is higher.

I have to strongly disagree: as we use a map (in the YAML manifest) which keys are the scmids: you cannot have 2 times the same id. If you have,n then something went terribly wrong. So the risk of 2 different SCM colliding is absolute zero (if you name the temp. dir with the scm id).

About the risk of collision with other objects: it means that the temp dir is falt with a list of ALL the temp dir that could be created. If you see a risk of collision due to naming, then it means that you have to namespace, otherwise you'll have a nasty bug hard to debug quite soon (even outisde this PR).
WDYT to namespace with the resource type (either with subdirectories or part of the name)? 

eg. `${TMP_DIR}/updatecli/scms/example1/` and `${TMP_DIR}/updatecli/scms/example2/` in your example?
(and `${TMP_DIR}/updatecli/source/example1/` , `${TMP_DIR}/updatecli/target/example2/`, etc.)?

> 
> Another option would be to use the same approach than the git repository so for example `https://github.com/updatecli/updatecli.git` becomes `/tmp/updatecli/https_github_com_updatecli_updatecli_git`

I find this example pretty hard to understand and read. It does not tell me immediatly what is this folder for. However it is an implementation detail= end user will never check it.

Reply:
> The thing is by design, when scm repositories are cloned locally in the temporary directory, they are reused across the different pipelines. So that we don't have to delete them and re-clone them between each pipeline.
> Which means that using the scmid, doesn't tell if ${TMP_DIR}/updatecli/scms/example1/ is a scm of type github or gitea

I don't see why this behavior requires the "kind" of scm to be know in the path? 

Additionnaly, I consider the "cross pipeline caching" to be both a bug (I can't reliably determine the outcome of updatecli without having read updatecli source code) and a security issue (code injection through deterministic caching). 
I would expect caching to be namespaces per pipeline.